src/ImageSharp/Formats/IImageDecoder.cs

```diff
@@ -19,19 +19,21 @@ public interface IImageDecoder
         /// <typeparam name="TPixel">The pixel format.</typeparam>
         /// <param name="configuration">The configuration for the image.</param>
         /// <param name="stream">The <see cref="Stream"/> containing image data.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
         /// <returns>The <see cref="Image{TPixel}"/>.</returns>
         // TODO: Document ImageFormatExceptions (https://github.com/SixLabors/ImageSharp/issues/1110)
-        Image<TPixel> Decode<TPixel>(Configuration configuration, Stream stream)
+        Image<TPixel> Decode<TPixel>(Configuration configuration, Stream stream, CancellationToken cancellationToken = default)
```

Comment:
Since no users are expected to call this method directly, we should avoid making the `cancellationToken` argument optional.

Reply:
Really?  Is that a stylistic choice within the ImageSharp codebase not to have optional arguments?  Given that these are methods on a public interface, I would have assumed that there might be people using these APIs directly, so providing a default for the `cancellationToken` seemed like a good way to minimize the breaking nature of the change.

Reply:
Yeah we tend to avoid optional arguments, because there is higher chance for making binary breaking changes. (It can be a source of endless arguments whether this is right or not, but it's not the point here and `CancellationToken` is actually an exception for this rule.)

What's important in this context is that `IImageDecoder` is not meant to be consumed, but to *be implemented*. With this in mind, avoiding `cancellationToken = default` makes things simpler.